test/unit/cli/test_option_mapper.py

```diff
@@ -0,0 +1,175 @@
+from dataclasses import dataclass
+from pathlib import Path
+from test.unit.cli.scs_mock import (
+    ScsMock,
+    ScsPatcher,
+)
+
+import pytest
+
+from exasol.nb_connector.ai_lab_config import AILabConfig as CKey
+from exasol.nb_connector.ai_lab_config import StorageBackend
+from exasol.nb_connector.cli.options import (
+    DOCKER_DB_OPTIONS,
+    ONPREM_OPTIONS,
+    SAAS_OPTIONS,
+)
+from exasol.nb_connector.cli.param_wrappers import ScsArgument
+from exasol.nb_connector.cli.processing import option_set
+from exasol.nb_connector.cli.processing.option_set import (
+    SELECT_BACKEND_OPTION,
+    USE_ITDE_OPTION,
+    BackendSelector,
+    OptionSet,
+    ScsCliError,
+    get_option_set,
+)
+
+
+@pytest.fixture
+def scs_patcher(monkeypatch):
+    return ScsPatcher(monkeypatch, module=option_set)
+
+
+@dataclass
+class Scenario:
+    name: str
+    backend: StorageBackend
+    use_itde: bool
+    params: list[ScsArgument]
+
+
+TEST_SCENARIOS = [
+    Scenario("SaaS", StorageBackend.saas, True, SAAS_OPTIONS),
+    Scenario("SaaS", StorageBackend.saas, False, SAAS_OPTIONS),
+    Scenario("Docker", StorageBackend.onprem, True, DOCKER_DB_OPTIONS),
+    Scenario("on-premise", StorageBackend.onprem, False, ONPREM_OPTIONS),
+]
+
+
+def test_empty_backend_configuration():
+    testee = BackendSelector(ScsMock())
+    assert not testee.knows_backend
+    assert not testee.knows_itde_usage
+    assert not testee.is_valid
+    for s in TEST_SCENARIOS:
+        assert testee.allows(s.backend, s.use_itde)
+
+
+@pytest.mark.parametrize(
+    "backend, use_itde",
+    [
+        (StorageBackend.saas, None),
+        (StorageBackend.onprem, None),
+        (None, True),
+        (None, False),
+    ],
+)
+def test_partial_backend_configuration(backend, use_itde):
+    scs_mock = ScsMock(backend, use_itde)
+    testee = BackendSelector(scs_mock)
+    assert testee.knows_backend == (backend is not None)
+    assert testee.knows_itde_usage == (use_itde is not None)
+    assert not testee.is_valid
+    for s in TEST_SCENARIOS:
+        assert testee.allows(s.backend, s.use_itde)
+
+
+@pytest.mark.parametrize("scenario", TEST_SCENARIOS)
+def test_valid_backend_configuration(scenario):
+    scs_mock = ScsMock(scenario.backend, scenario.use_itde)
+    testee = BackendSelector(scs_mock)
+    assert testee.knows_backend
+    assert testee.knows_itde_usage
+    assert testee.backend_name == scenario.name
+    assert testee.use_itde == scenario.use_itde
+    assert testee.is_valid
+    for s in TEST_SCENARIOS:
+        expected = scenario == s
+        assert testee.allows(s.backend, s.use_itde) == expected
```

Comment:
This assumes that all scenarios have different `backend` and `use_itde`.

Reply:
I think there is still a misunderstanding in the semantics between options as compare to a _pure_ backend selection. Let's talk next week?

Reply:
Ah, yes, I agree, see latest push, please.